packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -0,0 +1,185 @@
+      const { USDC } = brands;
+      const proposalShapes = makeProposalShapes({ USDC, PoolShares });
+      const dust = AmountMath.make(USDC, 1n);
+      const shareWorth = makeParity(dust, PoolShares);
+      const { zcfSeat: poolSeat } = zcf.makeEmptySeatKit();
+      const shareWorthRecorderKit = tools.makeRecorderKit(
+        node,
+        /** @type {TypedPattern<ShareWorth>} */ (RatioShape),
+      );
+      return {
+        shareMint,
+        shareWorth,
+        poolSeat,
+        PoolShares,
+        proposalShapes,
+        shareWorthRecorderKit,
+      };
+    },
+    {
+      feeSink: {
+        /**
+         * @param {Amount<'nat'>} amount
+         * @param {Payment<'nat'>} payment
+         */
+        async receive(amount, payment) {
+          const { poolSeat, shareWorth } = this.state;
+          const { external } = this.facets;
+          await depositToSeat(
+            zcf,
+            poolSeat,
+            harden({ USDC: amount }),
+            harden({ USDC: payment }),
+          );
+          await external.setShareWorth(withFees(shareWorth, amount));
+        },
+      },
+
+      external: {
+        /** @param {ShareWorth} shareWorth */
+        async setShareWorth(shareWorth) {
+          const { recorder } = this.state.shareWorthRecorderKit;
+          this.state.shareWorth = shareWorth;
+          await recorder.write(shareWorth);
+        },
+      },
+
+      depositHandler: {
+        /** @param {ZCFSeat} lp */
+        async handle(lp) {
+          const { shareWorth, shareMint, poolSeat } = this.state;
+          const { external } = this.facets;
+
+          /** @type {USDCProposalShapes['deposit']} */
+          // @ts-expect-error ensured by proposalShape
+          const proposal = lp.getProposal();
+          const post = depositCalc(shareWorth, proposal);
+          // COMMIT POINT
+          const mint = shareMint.mintGains(post.payouts);
+          zcf.atomicRearrange(
+            harden([
+              [lp, poolSeat, proposal.give],
+              [mint, lp, post.payouts],
+            ]),
+          );
+          lp.exit();
+          mint.exit();
+          await external.setShareWorth(post.shareWorth);
+        },
+      },
+      withdrawHandler: {
+        /** @param {ZCFSeat} lp */
+        async handle(lp) {
+          const { shareWorth, shareMint, poolSeat } = this.state;
+          const { external } = this.facets;
+
+          /** @type {USDCProposalShapes['withdraw']} */
+          // @ts-expect-error ensured by proposalShape
+          const proposal = lp.getProposal();
+          const { zcfSeat: burn } = zcf.makeEmptySeatKit();
+          const post = withdrawCalc(shareWorth, proposal);
+          // COMMIT POINT
+          zcf.atomicRearrange(
+            harden([
+              [lp, burn, proposal.give],
+              [poolSeat, lp, post.payouts],
+            ]),
+          );
+          shareMint.burnLosses(proposal.give, burn);
```

Comment:
ditto re: minting, the shareWorth has changed at this point, even if seat adjustments fail somehow